modules/xml/xml.c

```diff
@@ -40,29 +46,42 @@ remove_trailing_dot(gchar *str)
     str[strlen(str)-1] = 0;
 }
 
+static void
+encode_and_append_value(GString *result, const gchar *current_value, gssize current_value_len)
+{
+  if (result->len > 0)
+    g_string_append_c(result, ',');
+  str_repr_encode_append(result, current_value, current_value_len, ",");
```

Comment:
Not the scope for this PR, but I wonder if we really need to escape the special characters (`"\b\f\n\r\t\\ "`), defined in `str_repr_encode_append()` for lists.

Another finding:
```
if (result->len > 0)
  g_string_append_c(result, ',');
str_repr_encode_append(result, additinal_value, len, ",")
```
is used for list element handling in other places as well, maybe we could give a `list_append` function, which would only escape `,` and would do the "check whether this is the first element or not" as well. Actually, that function would look something like `encode_and_append_value()`, so it could be enough to move this function out and be used by the other source codes.
Source codes, where I found similar behavior:
```
lib/logmsg/logmsg.c
lib/scanner/kv-scanner/kv-scanner.c
modules/basicfuncs/context-funcs.c
modules/basicfuncs/list-funcs.c
modules/xml/xml.c
```

Reply:
We discussed this in person, thanks for summarizing!
- should `str_repr_encode` quote our string in case of a whitespace, while the only special character in our lists is the comma characeter is a good point. Right now I don't think in this context it is needed.
- adding a higher level abstraction for list-handling is a good idea, I guess there were not many users before. Now kv-parser(), tags-parser(), xml() parser. 
- as you started with, it has to be defined what should be escaped in lists in general, before creating such API.